lib/auth/init.go

```diff
@@ -493,6 +500,182 @@ func migrateLegacyResources(ctx context.Context, cfg InitConfig, asrv *Server) e
+	}
+
+	migratedTcs, err := migrateOSSTrustedClusters(ctx, role, asrv)
+	if err != nil {
+		return trace.Wrap(err, migrationAbortedMessage)
+	}
+
+	migratedConns, err := migrateOSSGithubConns(ctx, role, asrv)
+	if err != nil {
+		return trace.Wrap(err, migrationAbortedMessage)
+	}
+
+	if createdRoles > 0 || migratedUsers > 0 || migratedTcs > 0 || migratedConns > 0 {
+		log.Infof("Migration completed. Created %v roles, updated %v users, %v trusted clusters and %v Github connectors.",
+			createdRoles, migratedUsers, migratedTcs, migratedConns)
+	}
+
+	return nil
+}
+
+const remoteWildcardPattern = "^.+$"
+
+// migrateOSSTrustedClusters updates role mappings in trusted clusters
+// OSS Trusted clusters had no explicit mapping from remote roles, to local roles.
+// Map all remote roles to local OSS user role.
+func migrateOSSTrustedClusters(ctx context.Context, role types.Role, asrv *Server) (int, error) {
+	migratedTcs := 0
+	tcs, err := asrv.GetTrustedClusters()
+	if err != nil {
+		return migratedTcs, trace.Wrap(err, migrationAbortedMessage)
+	}
+
+	for _, tc := range tcs {
+		meta := tc.GetMetadata()
+		_, ok := meta.Labels[teleport.OSSMigratedV6]
+		if ok {
+			continue
+		}
+		setLabels(&meta.Labels, teleport.OSSMigratedV6, types.True)
+		roleMap := []types.RoleMapping{{Remote: remoteWildcardPattern, Local: []string{role.GetName()}}}
+		tc.SetRoleMap(roleMap)
+		tc.SetMetadata(meta)
+		if _, err := asrv.Presence.UpsertTrustedCluster(ctx, tc); err != nil {
+			return migratedTcs, trace.Wrap(err, migrationAbortedMessage)
+		}
+		for _, catype := range []services.CertAuthType{services.UserCA, services.HostCA} {
+			ca, err := asrv.GetCertAuthority(services.CertAuthID{Type: catype, DomainName: tc.GetName()}, true)
+			if err != nil {
+				return migratedTcs, trace.Wrap(err, migrationAbortedMessage)
+			}
+			meta := ca.GetMetadata()
+			_, ok := meta.Labels[teleport.OSSMigratedV6]
+			if ok {
+				continue
+			}
+			setLabels(&meta.Labels, teleport.OSSMigratedV6, types.True)
+			ca.SetRoleMap(roleMap)
+			ca.SetMetadata(meta)
+			err = asrv.UpsertCertAuthority(ca)
+			if err != nil {
+				return migratedTcs, trace.Wrap(err, migrationAbortedMessage)
+			}
+		}
+		migratedTcs++
+	}
+	return migratedTcs, nil
+}
+
+// migrateOSSUsers assigns all OSS users to a less privileged role
+// All OSS users were using implicit admin role. Migrate all users to less privileged
+// role that is read only and only lets users use assigned logins.
+func migrateOSSUsers(ctx context.Context, role types.Role, asrv *Server) (int, error) {
+	migratedUsers := 0
+	users, err := asrv.GetUsers(true)
+	if err != nil {
+		return migratedUsers, trace.Wrap(err, migrationAbortedMessage)
+	}
+
+	for _, user := range users {
+		meta := user.GetMetadata()
+		_, ok := meta.Labels[teleport.OSSMigratedV6]
+		if ok {
+			continue
+		}
+		setLabels(&meta.Labels, teleport.OSSMigratedV6, types.True)
+		user.SetRoles([]string{role.GetName()})
```

Comment:
This means OSS users will not have access to the Audit Log by default anymore right, because the new OSS default role does not have `KindEvents` and `KindSessions`? 

We should update documentation and release notes that when you upgrade all OSS users will lose the ability to view audit log and how to add it back.

Reply:
https://github.com/gravitational/teleport/pull/5419/files#diff-9b3fca74b2b2d4ce46a466de333f6c3e7640c21da8bd587e1837aa7a113eb7e7R221 

OSS Users will still be able to read events by default

Reply:
Got it, so they will be able to see the audit log but not view session recordings right? At least that's what I see from running this PR.